src/main/kotlin/org/dashevo/dapiclient/DapiClient.kt

```diff
@@ -1,219 +1,398 @@
+            return if(!serializedIdentityBinaryArray.isEmpty)
+                serializedIdentityBinaryArray
+            else null
+        } catch (e: StatusRuntimeException) {
+            if (e.status.code == Status.NOT_FOUND.code) {
+                return null;
+            }
+            throw e;
+        } finally {
+            shutdownInternal()
+        }
     }
 
     /**
-     * Send State Transition.
-     * @param stPacket: data packet.
-     * @param userRegTxId: Transaction ID of User creating the object.
+     * Fetch Data Contract by id
+     * @param contractId String
+     * @return ByteString? The contract bytes or null if not found
      */
-    fun sendStateTransition(stPacket: STPacket, userRegTxId: Sha256Hash, hashPrevSubTx: Sha256Hash, privKey: ECKey,
-                            cb: DapiRequestCallback<String>) {
+    fun getDataContract(contractId: String): ByteString? {
+        val getDataContractRequest = PlatformOuterClass.GetDataContractRequest.newBuilder()
+                .setId(contractId)
+                .build()
 
-        val serializedPacket = stPacket.serialize()
-        val stPacketHash = Sha256Hash.wrap(stPacket.hash())
+        val service = getPlatformService()
 
-        val stateTransitionTx = SubTxTransition(1, userRegTxId, hashPrevSubTx, Coin.valueOf(1000),
-                stPacketHash, privKey)
+        try {
+            var getDataContractResponse = service.getDataContract(getDataContractRequest);
+            return getDataContractResponse.dataContract ?: return null
+        } catch (e: StatusRuntimeException) {
+            if (e.status.code == Status.NOT_FOUND.code) {
+                return null
+            } else throw e;
+        } finally {
+            shutdownInternal()
+        }
+    }
 
-        val serializedStHex = stateTransitionTx.unsafeBitcoinSerialize().toHexString()
-        val serializedPacketHex = serializedPacket.toHexString()
+    /**
+     *
+     * @param contractId String The contract id associated with the documents
+     * @param type String The type of document
+     * @param documentQuery DocumentQuery DocumentQuery that specify which documents to find, sort order
+     * and pagination
+     * @return List<ByteArray>? a list of documents matching the provided parameters
+     */
+    fun getDocuments(contractId: String, type: String, documentQuery: DocumentQuery): List<ByteArray>? {
 
-        sendRawTransition(serializedStHex, serializedPacketHex, object : DapiRequestCallback<String> {
-            override fun onSuccess(data: JsonRPCResponse<String>) {
-                cb.onSuccess(data)
-            }
+        val builder = PlatformOuterClass.GetDocumentsRequest.newBuilder()
+                .setDataContractId(contractId)
+                .setDocumentType(type)
+                .setWhere(ByteString.copyFrom(documentQuery.encodeWhere()))
+                .setOrderBy(ByteString.copyFrom(documentQuery.encodeOrderBy()))
+        if(documentQuery.hasLimit())
```

Comment:
Would you mind running `Code -> Reformat Code` to keep the same formatting along the project?

Reply:
yes, I will do that.

Reply:
Reformatted here: 4a5d2f4